common/compiler-input.ts

```diff
@@ -0,0 +1,261 @@
+  stopAfter?: 'parsing'
+  /** Sorted list of remappings */
+  remappings?: string[]
+  /** Optimizer settings */
+  optimizer?: Partial<CompilerOptimizer>
+  /** Version of the EVM to compile for. Affects type checking and code generation */
+  evmVersion: EvmVersion;
+  /**
+   * Change compilation pipeline to go through the Yul intermediate representation.
+   * This is a highly EXPERIMENTAL feature, not to be used for production. This is false by default.
+   */
+  viaIR?: boolean
+  /** Debugging settings */
+  debug?: CompileDebug
+  /** Metadata settings */
+  metadata?: CompilerMetadata
+  /** Addresses of the libraries. If not all libraries are given here, it can result in unlinked objects whose output data is different. */
+  libraries: CompilerLibraries
+  /**
+   * The following can be used to select desired outputs.
+   * If this field is omitted, then the compiler loads and does type checking, but will not generate any outputs apart from errors.
+   * The first level key is the file name and the second is the contract name, where empty contract name refers to the file itself,
+   * while the star refers to all of the contracts.
+   * Note that using a using `evm`, `evm.bytecode`, `ewasm`, etc. will select every
+   * target part of that output. Additionally, `*` can be used as a wildcard to request everything.
+   * Contract level (needs the contract name or "*"):
+   * - abi - ABI
+   * - devdoc - Developer documentation (natspec)
+   * - userdoc - User documentation (natspec)
+   * - metadata - Metadata
+   * - ir - Yul intermediate representation of the code before optimization
+   * - irOptimized - Intermediate representation after optimization
+   * - storageLayout - Slots, offsets and types of the contract's state variables.
+   * - evm.assembly - New assembly format
+   * - evm.legacyAssembly - Old-style assembly format in JSON
+   * - evm.bytecode.functionDebugData - Debugging information at function level
+   * - evm.bytecode.object - Bytecode object
+   * - evm.bytecode.opcodes - Opcodes list
+   * - evm.bytecode.sourceMap - Source mapping (useful for debugging)
+   * - evm.bytecode.linkReferences - Link references (if unlinked object)
+   * - evm.bytecode.generatedSources - Sources generated by the compiler
+   * - evm.deployedBytecode* - Deployed bytecode (has all the options that evm.bytecode has)
+   * - evm.deployedBytecode.immutableReferences - Map from AST ids to bytecode ranges that reference immutables
+   * - evm.methodIdentifiers - The list of function hashes
+   * - evm.gasEstimates - Function gas estimates
+   * - ewasm.wast - Ewasm in WebAssembly S-expressions format
+   * - ewasm.wasm - Ewasm in WebAssembly binary format
+   */
+  outputSelection?: CompilerOutputSelection
+  /** The modelChecker object is experimental and subject to changes. */
+  modelChecker?: CompilerModelChecker
+}
+
+export interface CompilationInput {
```

Comment:
These definitions are pretty detailed. There are some problems though:
1) If we keep them here, they will be hard to keep in sync with the compiler. Maybe there are better ways to go about this? For example defining some JSON schema (not tied specifically to TypeScript syntax) that would be kept in the main compiler repo and only imported here?
2) solc-js is meant to support all the old compiler versions and Standard JSON changed over time. So these definitions won't be right for all the compiler binaries that you can use. So we either need separate definitions for different versions (at least breaking ones) or some superset that won't exactly fit any particular version.